File: SourceCode/SQLAPI/POS.Data/Entities/Supplier/Supplier.cs
using System;
using System.ComponentModel.DataAnnotations.Schema;

namespace POS.Data
{
    public class Supplier : BaseEntity
    {
        public Guid Id { get; set; }
        public string SupplierName { get; set; }
        public string ContactPerson { get; set; }
        public string Email { get; set; }
        public string Fax { get; set; }
        public string MobileNo { get; set; }
        public string PhoneNo { get; set; }
        public string Website { get; set; }
        public string Description { get; set; }
        public string Url { get; set; }
        public Guid BillingAddressId { get; set; }
        [ForeignKey("BillingAddressId")]
        public SupplierAddress BillingAddress { get; set; }
        public Guid ShippingAddressId { get; set; }
        [ForeignKey("ShippingAddressId")]
        public SupplierAddress ShippingAddress { get; set; }
        public string TaxNumber { get; set; }
      
    }
}

File: SourceCode/Angular/src/app/core/domain-classes/supplier.ts
import { EntityState } from './entity-state';
import { ContactAddress } from './contact-address';

export interface Supplier {
  id: string;
  supplierName: string;
  contactPerson: string;
  mobileNo: string;
  phoneNo: string;
  objectState?: EntityState;
  isDeleted?: boolean;
  isVerify?: boolean;
  isSendMail?: boolean;
  billingAddress?: ContactAddress | null;
  shippingAddress?: ContactAddress | null;
  description: string;
  website: string;
  url?: string;
  imageUrl?: string;
  logo?: string;
  isImageUpload?: boolean;
  email?: string;
  taxNumber?: string;
  billingAddressId?: string;
  shippingAddressId?: string;
}
